<a href="https://colab.research.google.com/github/OlhaZahrebelna/my_practis_on_github/blob/main/Zahrebelna_Olha_%22hw_6_2_%D0%A1%D1%82%D0%B0%D1%82%D0%B8%D1%81%D1%82%D0%B8%D1%87%D0%BD%D1%96_%D1%82%D0%B5%D1%81%D1%82%D0%B8_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [38]:
import numpy as np
import pandas as pd
from statsmodels.stats.weightstats import ztest

In [39]:
df = pd.read_csv('/content/sample_data/data.csv')
data = np.array(df['data'])

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [40]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [41]:
np.mean(data)


19.37755319670441

In [42]:
ztest_Score, p_value= ztest(data, value=20)

alpha = 0.05

print(f"z-статистика: {ztest_Score}")
print(f"p-value: {p_value}")

z-статистика: -0.9953477389335053
p-value: 0.31956716690719744


In [43]:
if p_value < alpha:
    print("Висновок: Статистично значуща різниця є.")
else:
    print("Висновок: Статистично значущої різниці немає.")

Висновок: Статистично значущої різниці немає.


Cереднє значення вибірки менше(z-статистика: -0.995), ніж гіпотетичне середнє.Не можна відхилити нульову гіпотезу, бо p-value: 0.3.

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [44]:
df_june = pd.read_csv('/content/data_2.csv')
data_june = np.array(df_june['data'])

In [45]:
data_june.shape

(120,)

In [46]:
np.mean(data_june), np.mean(data)

(19.37322276512519, 19.37755319670441)

In [47]:
value, p_value = ztest(data, data_june, value=0)

print(f"z-статистика: {value}")
print(f"p-value: {p_value}")

z-статистика: 0.005246673456808504
p-value: 0.9958137794593356


In [48]:
if p_value < alpha:
    print("Висновок: Статистично значуща різниця є.")
else:
    print("Висновок: Статистично значущої різниці немає.")

Висновок: Статистично значущої різниці немає.


Значення статистичного тесту та співвідношення між двома середніми значеннями збігаються. Це означає, що між середнім чеком за обидва місяці немає суттєвої різниці.